framework/moose.mk

```diff
@@ -317,6 +317,10 @@ $(moose_revision_header): $(moose_HEADER_deps)
 	@echo "Checking if header needs updating: "$@"..."
 	$(shell $(FRAMEWORK_DIR)/scripts/get_repo_revision.py $(FRAMEWORK_DIR) \
 	  $(moose_revision_header) MOOSE)
+        # make sure the header generation step didn't fail
+	@if [ $(.SHELLSTATUS) -ne 0 ]; then \
```

Comment:
we're good on forcing GNU Make >=4.2 ?
@loganharbour

Reply:
Hmm... I think this is fine. Casually requiring six year old make.